src/deque.cr

```diff
@@ -581,15 +581,39 @@ class Deque(T)
     end
   end
 
-  private def increase_capacity
+  private INITIAL_CAPACITY = 4
+
+  # behaves like `calculate_new_capacity(@capacity + 1)`
+  private def calculate_new_capacity
+    return INITIAL_CAPACITY if @capacity == 0
+
+    @capacity * 2
+  end
+
+  private def calculate_new_capacity(new_size)
+    new_capacity = @capacity == 0 ? INITIAL_CAPACITY : @capacity
+    while new_capacity < new_size
+      new_capacity *= 2
+    end
+    new_capacity
+  end
+
+  private def resize_if_cant_insert(insert_size = 1)
+    new_capacity = calculate_new_capacity(@size + insert_size)
+    if new_capacity > @capacity
+      resize_to_capacity(new_capacity)
+    end
+  end
+
+  private def resize_to_capacity(capacity)
     unless @buffer
-      @capacity = 4
+      @capacity = capacity
       @buffer = Pointer(T).malloc(@capacity)
       return
     end
 
     old_capacity = @capacity
-    @capacity *= 2
+    @capacity = capacity
```

Comment:
Maybe this assignment could be unified with the one in the branch above and placed at the start of the method?